atomics/src/Lock_Array_CUDA.cpp

```diff
@@ -38,18 +40,36 @@ namespace Impl {
 int32_t* CUDA_SPACE_ATOMIC_LOCKS_DEVICE_h = nullptr;
 int32_t* CUDA_SPACE_ATOMIC_LOCKS_NODE_h = nullptr;
 
+void check_error_and_throw_cuda(cudaError e, const std::string msg) {
```

Comment:
Define it in the anonymous namespace

Reply:
done

Reply:
Why the anonymous namespace rather than inline?

Reply:
So it has internal linkage

Reply:
That's kinda why I'm asking, why do we want to have multiple copies of this in a final binary rather than letting the linker throw all but one away?

Reply:
I didn't want a symbol in the library and thought the compiler was likely to decide to inline by himself.
Are you ok if add inline and keep it in the anonymous namespace?